Sources/BSON/ExtendedJSONEncoder.swift

```diff
@@ -0,0 +1,29 @@
+import Foundation
+/// Facilitates the encoding of `Encodable` values into ExtendedJSON.
+public class ExtendedJSONEncoder {
+    internal enum extendedJSONMode {
+        case canonical
+        case relaxed
+    }
```

Comment:
Not sure If adding this enum is technically part of sketching the public API

Reply:
discussed offline: this is part of the public API--it's the enum users can set to determine whether the encoder produces relaxed or canonical extended JSON

Reply:
done